unity-client/Assets/Rendering/Culling/CullingController.cs

```diff
@@ -308,9 +307,15 @@ internal IEnumerator ProcessAnimations()
                 Transform t = anim.transform;
 
                 Vector3 playerPosition = CommonScriptableObjects.playerUnityPosition;
-                float distance = Vector3.Distance(playerPosition, t.position);
 
-                if (distance > settings.enableAnimationCullingDistance)
+                Bounds bounds = anim.localBounds;
+                bounds.center += t.position;
+
+                Vector3 boundingPoint = bounds.ClosestPoint(playerPosition);
+                float distance = Vector3.Distance(playerPosition, boundingPoint);
+                bool boundsContainsPlayer = bounds.Contains(playerPosition);
+
+                if (distance > settings.enableAnimationCullingDistance && !boundsContainsPlayer)
```

Comment:
After all we were checking the distance against the pivot. 

Now the culling works if:
- The distance against the animation bounding box nearest point is above a limit
- The player is not inside the bounds. i.e.: If the player is inside the bounds the animation will ALWAYS be played.